crates/trie/parallel/src/proof_task.rs

```diff
@@ -287,55 +403,139 @@ pub struct ProofTaskCtx {
         nodes_sorted: Arc<TrieUpdatesSorted>,
         state_sorted: Arc<HashedPostStateSorted>,
+        prefix_sets: Arc<TriePrefixSetsMut>,
     ) -> Self {
-        Self { nodes_sorted, state_sorted }
+        Self { nodes_sorted, state_sorted, prefix_sets }
     }
 }
 
-/// A message used to send a storage proof request to the [`ProofTaskManager`].
+/// Message used to communicate with [`ProofTaskManager`].
 #[derive(Debug)]
 pub enum ProofTaskMessage<Tx> {
-    /// A storage proof request.
-    StorageProof((StorageProofInput, Sender<StorageProofResult>)),
+    /// A request to queue a proof task.
+    QueueTask(ProofTaskKind),
     /// A returned database transaction.
     Transaction(ProofTaskTx<Tx>),
     /// A request to terminate the proof task manager.
     Terminate,
 }
 
-/// A handle that wraps a single proof task sender that sends a terminate message on `Drop`.
+/// Proof task kind.
+///
+/// When queueing a task using [`ProofTaskMessage::QueueTask`], this enum
+/// specifies the type of proof task to be executed.
+#[derive(Debug)]
+pub enum ProofTaskKind {
+    /// A storage proof request.
+    StorageProof(StorageProofInput, Sender<StorageProofResult>),
+    /// A blinded account node request.
+    BlindedAccountNode(Nibbles, Sender<BlindedNodeResult>),
+    /// A blinded storage node request.
+    BlindedStorageNode(B256, Nibbles, Sender<BlindedNodeResult>),
+}
+
+/// A handle that wraps a single proof task sender that sends a terminate message on `Drop` if the
+/// number of active handles went to zero.
 #[derive(Debug)]
 pub struct ProofTaskManagerHandle<Tx> {
     /// The sender for the proof task manager.
     sender: Sender<ProofTaskMessage<Tx>>,
+    /// The number of active handles.
+    active_handles: Arc<AtomicUsize>,
 }
 
 impl<Tx> ProofTaskManagerHandle<Tx> {
     /// Creates a new [`ProofTaskManagerHandle`] with the given sender.
-    pub const fn new(sender: Sender<ProofTaskMessage<Tx>>) -> Self {
-        Self { sender }
+    pub fn new(sender: Sender<ProofTaskMessage<Tx>>, active_handles: Arc<AtomicUsize>) -> Self {
+        active_handles.fetch_add(1, Ordering::SeqCst);
+        Self { sender, active_handles }
     }
 
-    /// Clones and returns the inner sender.
-    pub fn sender(&self) -> Sender<ProofTaskMessage<Tx>> {
-        self.sender.clone()
+    /// Queues a task to the proof task manager.
+    pub fn queue_task(&self, task: ProofTaskKind) -> Result<(), SendError<ProofTaskMessage<Tx>>> {
+        self.sender.send(ProofTaskMessage::QueueTask(task))
     }
 
-    /// Sends a terminate message to the proof task manager.
+    /// Terminates the proof task manager.
     pub fn terminate(&self) {
         let _ = self.sender.send(ProofTaskMessage::Terminate);
     }
 }
 
+impl<Tx> Clone for ProofTaskManagerHandle<Tx> {
+    fn clone(&self) -> Self {
+        self.active_handles.fetch_add(1, Ordering::SeqCst);
+        Self { sender: self.sender.clone(), active_handles: self.active_handles.clone() }
```

Comment:
this is just `Self::new(clones)`

we should minimize where we deal with the counter